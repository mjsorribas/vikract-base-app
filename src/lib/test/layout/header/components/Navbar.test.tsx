import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Navbar from "lib/layout/header/components/navbar/NavBar";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test("Navbar", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <Navbar />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
