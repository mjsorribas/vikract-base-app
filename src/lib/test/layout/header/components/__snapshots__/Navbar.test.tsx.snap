// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar 1`] = `
<div
  className="css-0"
>
  <div
    className="css-k008qs"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-0"
      >
        <img
          className="chakra-image css-1chmwhm"
          src="/assets/vikract.svg"
          title="The Vikract Webapp template"
        />
        <p
          className="chakra-text css-104caxi"
        >
          THE VIKRACT TEMPLATE
        </p>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-70qvj9"
    >
      <div
        className="chakra-stack css-ex90l4"
      >
        <a
          className="chakra-link css-177s7ri"
          href="/"
        >
          Home
        </a>
        <a
          className="chakra-link css-1jpe0vt"
          href="/aboutus"
        >
          About Us
        </a>
        <a
          className="chakra-link css-1jpe0vt"
          href="/prices"
        >
          Prices
        </a>
        <a
          className="chakra-link css-1jpe0vt"
          href="/contactus"
        >
          Contact Us
        </a>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-0"
    >
      <div
        className="css-0"
      >
        <div
          className="chakra-stack css-111rh5x"
        >
          <button
            className="chakra-button css-4xx2wk"
            type="button"
          >
            Sign Up
          </button>
          <button
            className="chakra-button css-4xx2wk"
            type="button"
          >
            Sign In
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Navbar Mobile true 1`] = `
<div
  className="css-0"
>
  <div
    className="css-k008qs"
  >
    <div
      className="css-70qvj9"
    >
      <div
        className="css-0"
      >
        <img
          className="chakra-image css-1chmwhm"
          src="/assets/vikract.svg"
          title="The Vikract Webapp template"
        />
        <p
          className="chakra-text css-104caxi"
        >
          THE VIKRACT TEMPLATE
        </p>
      </div>
    </div>
    <div
      className="css-17xejub"
    />
    <div
      className="css-70qvj9"
    >
      <button
        aria-controls="menu-list-:r9:"
        aria-expanded={false}
        aria-haspopup="menu"
        aria-label="Options"
        className="chakra-button chakra-menu__menu-button css-ymfqft"
        id="menu-button-:r9:"
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="currentColor"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 4H21V6H3V4ZM3 19H21V21H3V19ZM3 14H21V16H3V14ZM3 9H21V11H3V9Z"
          />
        </svg>
      </button>
      <div
        className="css-0"
        style={
          {
            "inset": "0 auto auto 0",
            "minWidth": "max-content",
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div
          aria-orientation="vertical"
          className="chakra-menu__menu-list css-luqtzj"
          id="menu-list-:r9:"
          onKeyDown={[Function]}
          role="menu"
          style={
            {
              "opacity": 0,
              "transform": "scale(0.8) translateZ(0)",
              "transformOrigin": "var(--popper-transform-origin)",
              "visibility": "hidden",
            }
          }
          tabIndex={-1}
        >
          <button
            className="chakra-menu__menuitem css-12st3z8"
            id="menu-list-:r9:-menuitem-:ra:"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            <a
              className="chakra-link css-f9i6ey"
              href="/"
            >
              Home
            </a>
          </button>
          <button
            className="chakra-menu__menuitem css-1k756t9"
            id="menu-list-:r9:-menuitem-:rb:"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            <a
              className="chakra-link css-1oxylfh"
              href="/aboutus"
            >
              About Us
            </a>
          </button>
          <button
            className="chakra-menu__menuitem css-1k756t9"
            id="menu-list-:r9:-menuitem-:rc:"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            <a
              className="chakra-link css-1oxylfh"
              href="/prices"
            >
              Prices
            </a>
          </button>
          <button
            className="chakra-menu__menuitem css-1k756t9"
            id="menu-list-:r9:-menuitem-:rd:"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            tabIndex={0}
            type="button"
          >
            <a
              className="chakra-link css-1oxylfh"
              href="/contactus"
            >
              Contact Us
            </a>
          </button>
          <hr
            aria-orientation="horizontal"
            className="chakra-menu__divider css-0"
          />
          <div
            className="css-0"
          >
            <div
              className="css-6ps7pe"
            >
              <button
                className="chakra-button css-10tmkll"
                type="button"
              >
                Sign Up
              </button>
              <button
                className="chakra-button css-1lxikdv"
                type="button"
              >
                Sign In
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
